bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,273 @@
+ */
+package org.openhab.binding.openthermgateway.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.OpenThermGatewayBindingConstants;
+import org.openhab.binding.openthermgateway.internal.DataItem;
+import org.openhab.binding.openthermgateway.internal.DataItemGroup;
+import org.openhab.binding.openthermgateway.internal.GatewayCommand;
+import org.openhab.binding.openthermgateway.internal.GatewayCommandCode;
+import org.openhab.binding.openthermgateway.internal.Message;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayCallback;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConfiguration;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConnector;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewaySocketConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenThermGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class OpenThermGatewayHandler extends BaseThingHandler implements OpenThermGatewayCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenThermGatewayHandler.class);
+
+    private @Nullable OpenThermGatewayConfiguration config;
+
+    private @Nullable OpenThermGatewayConnector connector;
+
+    private boolean connecting = false;
+
+    private boolean explicitDisconnect = false;
+
+    public OpenThermGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenTherm Gateway handler for uid '{}'", getThing().getUID());
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Initializing");
+
+        config = getConfigAs(OpenThermGatewayConfiguration.class);
+
+        connect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        if (!(command instanceof RefreshType)) {
```

Comment:
I personally would write this as kind of a pre-condition check so that the rest of the method does not need that much indentation:

```java
if (command instanceof RefreshType) {
   return;
}

... rest of the code